libs/filamat/src/MaterialBuilder.cpp

```diff
@@ -468,8 +474,13 @@ bool MaterialBuilder::runSemanticAnalysis() noexcept {
 
     TargetApi targetApi = params.targetApi;
     assertSingleTargetApi(targetApi);
-    ShaderModel model = static_cast<ShaderModel>(params.shaderModel);
 
+    if (mEnableFramebufferFetch) {
+        // framebuffer fetch is only available with vulkan semantics
+        targetApi = TargetApi::VULKAN;
+    }
```

Comment:
Why do we need to switch to `TargetApi::VULKAN` here, but not in other places, like in `findProperties` or `generateShaders`?

Reply:
honestly I think the existing code doesn't completely make sense in the first place -- so I just tried to make the smallest change possible.